crates/uv/src/lib.rs

```diff
@@ -42,8 +44,45 @@ pub(crate) mod printer;
 pub(crate) mod settings;
 pub(crate) mod version;
 
+/// Resolves the script target for a run command.
+/// If it's a local file, does nothing. If it's a URL, downloads the content
+/// to a temporary file and updates the command. Prioritizes local files over URLs.
+/// Returns Some(NamedTempFile) if a remote script was downloaded, None otherwise.
+async fn resolve_script_target(
+    command: &mut ExternalCommand,
+) -> Result<Option<tempfile::NamedTempFile>> {
+    let Some(target) = command.get_mut(0) else {
+        return Ok(None);
+    };
+
+    if Path::new(target).is_file() {
```

Comment:
I think you should be using [`Path::try_exists`](https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists) here. If it's result is anything other than `Ok(true)`, then you bail.

The idea here is that I think we want to be pretty conservative: we should only execute a remote script if we can _absolutely confirm_ there is no corresponding local file path. Otherwise, you open yourself up to other vectors where by an end user unintentionally runs a remote script.

Reply:
> I think you should be using `Path::try_exists` here.

Thanks for the pointer!

> If it's result is anything other than `Ok(true)`, then you bail.

Maybe I'm having a hard time thinking this through, but right we "bail" from this function if a file exists. I think we want to bail for anything that is not `Ok(false)`? (I went ahead and made these changes, sorry if I'm missing something).

Reply:
Yes, you're 100% right. I got the logic flipped around! Whoops.